src/main/java/org/opensearch/knn/training/FloatTrainingDataConsumer.java

```diff
@@ -98,6 +98,12 @@ private List<byte[]> quantizeVectors(List<?> vectors) throws IOException {
             public float[] getVectorAtThePosition(int position) {
                 return ArrayUtils.toPrimitive((Float[]) vectors.get(position));
             }
+
+            @Override
+            public void resetVectorValues() {
+                // No-op
```

Comment:
Why no-op?

Reply:
Since `getVectorAtThePosition` doesn't change the state of the vectors. `resetVectorValues()` is in the interface since for `KNNVectorQuantizationTrainingRequest` we need `resetVectorValues()` to reset the `knnVectorValuesSupplier` iterator state.

I'll add a clarifying comment here.